libraries/AP_Proximity/AP_Proximity_Cygbot_D1.cpp

```diff
@@ -97,26 +97,24 @@ bool AP_Proximity_Cygbot_D1::parse_byte(uint8_t data)
             _parse_state = Payload_Data;
             _msg.payload_counter = 0;
             _msg.payload[_msg.payload_counter] = data;
-            _msg.payload_counter ++;
```

Comment:
It's a bit odd to have the payload_counter be zero but still use payload[0].  Doesn't this mean that there's no way for the payload to be empty?  I suspect it works it's just a little different from how we normally handle buffers.  Normally the counter or len parameter will be "1" when there is just a single element in the buffer at index 0.